paymentsheet/src/main/java/com/stripe/android/link/ui/updatecard/UpdateCardScreenViewModel.kt

```diff
@@ -2,26 +2,61 @@ package com.stripe.android.link.ui.updatecard
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
 import androidx.lifecycle.viewmodel.initializer
 import androidx.lifecycle.viewmodel.viewModelFactory
+import com.stripe.android.DefaultCardBrandFilter
 import com.stripe.android.core.Logger
+import com.stripe.android.link.account.LinkAccountManager
 import com.stripe.android.link.injection.NativeLinkComponent
+import com.stripe.android.model.ConsumerPaymentDetails
+import com.stripe.android.paymentsheet.CardUpdateParams
+import com.stripe.android.paymentsheet.PaymentSheet.BillingDetailsCollectionConfiguration.AddressCollectionMode
+import com.stripe.android.paymentsheet.ui.DefaultEditCardDetailsInteractor
+import com.stripe.android.paymentsheet.ui.EditCardDetailsInteractor
+import com.stripe.android.paymentsheet.ui.EditCardPayload
+import com.stripe.android.uicore.navigation.NavigationManager
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import javax.inject.Inject
 
 internal class UpdateCardScreenViewModel @Inject constructor(
     private val logger: Logger,
-    initialState: UpdateCardScreenState,
+    private val linkAccountManager: LinkAccountManager,
+    private val navigationManager: NavigationManager,
+    paymentDetailsId: String,
 ) : ViewModel() {
 
-    private val _state = MutableStateFlow(initialState)
-    val state: StateFlow<UpdateCardScreenState> = _state.asStateFlow()
+    private val _state = MutableStateFlow(UpdateCardScreenState())
+    val state: StateFlow<UpdateCardScreenState?> = _state.asStateFlow()
+
+    lateinit var interactor: EditCardDetailsInteractor
```

Comment:
Let's make this one nullable instead.

Reply:
https://github.com/stripe/stripe-android/pull/10715/commits/39a1b35696357c46c02106d9ea975fea48bab141